jobs/build/refresh-images/Jenkinsfile

```diff
@@ -112,15 +142,79 @@ images:update-dockerfile
+//
+                BUILD_CONTINUED = false
+                waitUntil {
+                    try {
+                        exclude = ""
+                        if (BUILD_EXCLUSIONS != "") {
+                            exclude = "-x ${BUILD_EXCLUSIONS} --ignore-missing-base"
+                        }
+
+                        buildlib.oit """
 --working-dir ${OIT_WORKING} --group openshift-${OSE_MAJOR}.${OSE_MINOR}
 images:build
+${exclude}
 --push-to-defaults --repo-type signed
 """
+                        return true  // finish waitUntil
+                    }
+
+                    catch (err) {
+                        failed_builds = get_failed_builds(OIT_WORKING)
+
+                        mail(
+                            to: "${MAIL_LIST_FAILURE}",
+                            from: "aos-cicd@redhat.com",
+                            subject: "RESUMABLE Error during Refresh Images for OCP v${OSE_MAJOR}.${OSE_MINOR}",
+                            body: """Encountered an error: ${err}
+Input URL: ${env.BUILD_URL}input
+Jenkins job: ${env.BUILD_URL}
 
-                try {
+BUILD / PUSH FAILURES:
+${failed_builds}
+""")
+
+                        def resp = input message: "Error during Image Build for OCP v${OSE_MAJOR}.${OSE_MINOR}",
+                        parameters: [
+                            [$class: 'hudson.model.ChoiceParameterDefinition',choices: "RETRY\nCONTINUE\nABORT", name: 'action', description: 'Retry (try the operation again). Continue (fails are OK, continue pipeline). Abort (terminate the pipeline).']
+                        ]
+
+                        switch (resp) {
+                            case "RETRY":
+                                // cause waitUntil to loop again
+                                return false
+                            case "CONTINUE":
+                                echo "User chose continue. Build failures are non-fatal."
+                                //will make email show PARTIAL, don't try to rebuild failures, only unbuilt
+                                BUILD_EXCLUSIONS = failed_builds.keySet().join(",")
+                                //simply setting flag to keep required work out of input flow
+                                BUILD_CONTINUED = true
+                                return true // Terminate waitUntil
+                            default: // ABORT
+                                error("User chose to abort pipeline because of image build failures")
+                        }
+
+                    }
+
+                }
+
+                // a failed build won't push. If continued, do that now
+                if (BUILD_CONTINUED) {
                     buildlib.oit """
 --working-dir ${OIT_WORKING} --group openshift-${OSE_MAJOR}.${OSE_MINOR}
+${exclude}
+images:push
+--to-defaults --late-only"""
+                    // exclude is already set earlier in the main images:build flow
+                }
+            }
+
+            try {
```

Comment:
`try {` and `} catch ...` look to be indented differently: https://github.com/markllama/aos-cd-jobs/blob/6a23f501ed6b6f5ed0010a13417d6ed97c3d2193/jobs/build/refresh-images/Jenkinsfile#L215-L221